test_conformance/extensions/cl_khr_command_buffer/command_buffer_profiling.cpp

```diff
@@ -0,0 +1,289 @@
+        }
+        else
+        {
+            log_info(
+                "Queue property CL_QUEUE_PROFILING_ENABLE not supported \n");
+            return CL_SUCCESS;
+        }
+
+        return BasicCommandBufferTest::SetUp(elements);
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int Run() override
+    {
+        cl_int error = CL_SUCCESS;
+
+        // record command buffer
+        error = RecordCommandBuffer();
+        test_error(error, "RecordCommandBuffer failed");
+
+        if (simultaneous_use_requested)
+        {
+            // enqueue simultaneous command-buffers with profiling command queue
+            error = RunSimultaneous();
+            test_error(error, "RunSimultaneous failed");
+        }
+        else
+        {
+            // enqueue single command-buffer with profiling command queue
+            error = RunSingle();
+            test_error(error, "RunSingle failed");
+        }
+
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int RecordCommandBuffer()
+    {
+        cl_int error = CL_SUCCESS;
+
+        error = clCommandNDRangeKernelKHR(
+            command_buffer, nullptr, nullptr, kernel, 1, nullptr, &num_elements,
+            nullptr, 0, nullptr, nullptr, nullptr);
+        test_error(error, "clCommandNDRangeKernelKHR failed");
+
+        error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+#define ADD_PROF_PARAM(prop)                                                   \
+    {                                                                          \
+        prop, #prop, 0                                                         \
+    }
+    struct ProfilingParam
+    {
+        cl_profiling_info param;
+        std::string name;
+        cl_ulong value;
+    };
+
+    //--------------------------------------------------------------------------
+    cl_int VerifyResult(const clEventWrapper& event)
+    {
+        cl_int error = CL_SUCCESS;
+        cl_int status;
+        error = clGetEventInfo(event, CL_EVENT_COMMAND_EXECUTION_STATUS,
+                               sizeof(status), &status, NULL);
+        test_error(error, "clGetEventInfo() failed");
+
+        if (status != CL_SUCCESS)
+            test_fail("Kernel execution status %d! (%s:%d)\n", status, __FILE__,
+                      __LINE__);
+
+        std::vector<ProfilingParam> prof_params = {
+            ADD_PROF_PARAM(CL_PROFILING_COMMAND_QUEUED),
+            ADD_PROF_PARAM(CL_PROFILING_COMMAND_SUBMIT),
+            ADD_PROF_PARAM(CL_PROFILING_COMMAND_START),
+            ADD_PROF_PARAM(CL_PROFILING_COMMAND_END),
+            ADD_PROF_PARAM(CL_PROFILING_COMMAND_COMPLETE)
```

Comment:
[CL_PROFILING_COMMAND_COMPLETE](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_PROFILING_COMMAND_COMPLETE) is only available after OpenCL version 2.0, so I don't think this will work with 1.2 implementations, which this extension should support.

The way you write version dependent tests in the CTS is through the `ADD_TEST_VERSION` macro (rather than doing `ADD_TEST`), e.g `ADD_TEST_VERSION(my_test, Version(2, 0))`.

However, I can't see any testing of `CL_PROFILING_COMMAND_COMPLETE` in the main `test_conformance/profiling` tests, so this is a more general gap in CTS testing and I'd be happy to just omit testing of `CL_PROFILING_COMMAND_COMPLETE` for the scope of this PR.

Reply:
Done, just for completeness: I used test device_execution -> enqueue_profiling as an example which uses this flag but verifies OpenCL 2.0 by separate initialization code:

`auto expected_min_version = Version(2, 0);`